enhancements/baremetal/enable-baremetal-on-aws.md

```diff
@@ -0,0 +1,150 @@
+
+Currently CBO checks the platform and if it is not baremetal, Openstack, vSphere, or None, it will be in a "disabled" state i.e. it will
+1. set status.conditions Disabled=true and
+2. not read or process the Provisioning CR and thus not deploy baremetal-operator.
+
+This proposal is to allow CBO to be enabled on AWS platform.
+
+To keep the testing matrix as it is, the allowed configuration options
+of the Provisioning CR will be kept restricted to exactly those required by centralized host management.
+
+*Only spec.provisioningNetwork=Disabled mode will still be accepted in the Provisioning CR.*
+
+If any other provisioningNetwork mode is set, the CBO webhook will refuse the change in the usual way.
+
+Notes:
+
+1. when the Provisioning CR is set to provisioningNetwork=Disabled mode, worker nodes would be booted via virtual media. This removes the requirement for the Provisioning Network which can be expected to be available only in Baremetal platform types.
+
+2. documentation will need to be added to the centralized host management documentation explaining how to create and update a Provisioning CR for the AWS platform.
+
+### Workflow Description
+
+#### Variation
+
+### User Stories
+
+#### Story 1 - Current IPI baremetal platform use case
+
+No change.
+
+#### Story 2 - centralized host management use case
+
+As a user of a hub cluster that performs central infrastructure management, and optionally zero-touch provisioning, I need to provision hosts using the k8s-native API (Baremetal Hosts CR) even when the hub cluster has a platform of AWS.
+
+### Risks and Mitigations
+
+One concern of this implementation is that for the solution to work properly the ip address of the BMCs of the bare metal hosts deployed and managed in this way will be exposed externally, as they need to communicate with the control plane, hosted in the AWS platform.
+It is essential to use some vpn/tunnel-like connection between the bare metal hosts and the control plane.
+For example, a native solution commonly used in AWS is the [VPC connection](https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html).
+A risk is given by the fact that we don't have a way to verify the correctness of this configuration when directly managed by the user.
+
+### Drawbacks
+
+## Design Details
+
+### API Extensions
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+#### Support Procedures
+
+### Test Plan
+
+#### Unit Testing
+
+We will add unit tests to confirm that cluster-baremetal-operator:
+* is enabled on the AWS platform
+* will restrict functionality on the AWS platform to ProvisioningNetork=Disabled
+
+#### Functional Testing
+
+QE will validate that the AWS platform is supported to reduce the load on CI.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+The feature is in Tech Preview since Openshift 4.12
```

Comment:
Is this actually enforced (i.e. requires TechPreviewNoUpgrade flag) or is it all on the honour system?

Reply:
this was not enforced, it was actually silently enabled

Reply:
I made it explicit that the feature was silently enabled in 4.12